### PR TITLE
Refer to #error pragma

### LINKED ISSUE #21563: #error version Tip Not Explicitly Clear
The tip about using '#error version' to determine the C# language version should make it explicitly clear that IntelliSense will flag the directive and the project will not build.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5694e62b-c756-1f62-4014-2ceafa54c011
* Version Independent ID: 3f01c378-22e4-197c-a862-ab4a924506c7
* Content: [C# language versioning - C# Guide](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version)
* Content Source: [docs/csharp/language-reference/configure-language-version.md](https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/configure-language-version.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by smholvoet**
Fixes #21563 🛠️

👉 Made it more clear that the compiler will report an error and referenced the [#error](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-error) preprocessor directive.

### CODE REVIEW

File: docs/csharp/language-reference/configure-language-version.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ If you must specify your C# version explicitly, you can do so in several ways:
 - Configure the [`-langversion` compiler option](compiler-options/langversion-compiler-option.md).
 
 > [!TIP]
-> To know what language version you're currently using, put `#error version` (case sensitive) in your code. This makes the compiler produce a diagnostic, CS8304, with a message containing the compiler version being used and the current selected language version.
+> To know what language version you're currently using, put `#error version` (case sensitive) in your code. This makes the compiler report a compiler error, CS8304, with a message containing the compiler version being used and the current selected language version. See [#error (C# Reference)](/preprocessor-directives/preprocessor-error) for more information.
```
Comment (Youssef1313) [inline docs/csharp/language-reference/configure-language-version.md:51]: Can you use a relative link here?

Comment (smholvoet) [reply docs/csharp/language-reference/configure-language-version.md]: Getting rid of the leading space should fix this, thanks for noticing 🕵️

File: docs/csharp/language-reference/configure-language-version.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ If you must specify your C# version explicitly, you can do so in several ways:
 - Configure the [`-langversion` compiler option](compiler-options/langversion-compiler-option.md).
 
 > [!TIP]
-> To know what language version you're currently using, put `#error version` (case sensitive) in your code. This makes the compiler produce a diagnostic, CS8304, with a message containing the compiler version being used and the current selected language version.
+> To know what language version you're currently using, put `#error version` (case sensitive) in your code. This makes the compiler report a compiler error, CS8304, with a message containing the compiler version being used and the current selected language version. See [#error (C# Reference)](preprocessor-directives/preprocessor-error) for more information.
```
Comment (Youssef1313) [inline docs/csharp/language-reference/configure-language-version.md:51]:
```suggestion
> To know what language version you're currently using, put `#error version` (case sensitive) in your code. This makes the compiler report a compiler error, CS8304, with a message containing the compiler version being used and the current selected language version. See [#error (C# Reference)](preprocessor-directives/preprocessor-error.md) for more information.
```

**Review by BillWagner — APPROVED** [review]
Thanks for writing this @smholvoet 

I've reviewed the changes, and I'll :shipit: now

**Pull request merged**